### ISSUE TITLE
add possibility to set the pyproject TOML from a custom path when running a command

### ISSUE THREAD

**Issue by DavidVujic** [python-poetry/poetry#7039]
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have searched the [FAQ](https://python-poetry.org/docs/faq/) and general [documentation](https://python-poetry.org/docs/) and believe that my question is not already covered.

## Feature Request

I think it would be a nice idea to make it possible for a Poetry command (the built-ins or from a plugin) to pass in the path to the actual `pyproject.toml` file.

Something like:
```shell
poetry build --path my_project/in/a/subfolder/pyproject.toml
```
This would be useful for scenarios when you have more than one project in a repository. With such a feature, you could run build commands from the repo top folder, instead of navigating down the directory structure.

I see two steps here:
1. make it possible to set the `pyproject` property of the `poetry` instance. A command could then do:
```python
self.poetry.pyproject = PyProjectTOML(custom_path)
```
2. add  a command option. The closest at hand would probably be the `build` command.

**Comment by neersighted** [comment]
Addressed by #6810 already as best I can tell.

**Comment by damascene** [comment]
So we can now use:
```
poetry build --directory my_project/in/a/subfolder/
```
or:
```
poetry build -C my_project/in/a/subfolder/
```
considering that `pyproject.toml` location is:
```
my_project/in/a/subfolder/pyproject.toml
```
?
